kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemMetadataProvider.java

```diff
@@ -534,6 +535,16 @@ static void writeDescriptor(
         throw new DatasetIOException("Cannot close", e);
       }
     }
+
+    //rename the file into its final location so that changes appear atomic
+      try {
+          fs.rename(tempDescriptorPath, descriptorPath);
```

Comment:
Another wrinkle in the FileContext route is that it even if hadoop-1 is dropped it doesn't look like webhdfs is supported..

When you switch to using it you get an an error building the project with a message like:

```
org.apache.hadoop.fs.UnsupportedFileSystemException: No AbstractFileSystem for scheme: webhdfs
	at org.apache.hadoop.fs.AbstractFileSystem.createFileSystem(AbstractFileSystem.java:152)
	at org.apache.hadoop.fs.AbstractFileSystem.get(AbstractFileSystem.java:240)
	at org.apache.hadoop.fs.FileContext$2.run(FileContext.java:332)
	at org.apache.hadoop.fs.FileContext$2.run(FileContext.java:329)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:415)
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1548)
	at org.apache.hadoop.fs.FileContext.getAbstractFileSystem(FileContext.java:329)
	at org.apache.hadoop.fs.FileContext.getFileContext(FileContext.java:443)
	at org.apache.hadoop.fs.FileContext.getFileContext(FileContext.java:423)
	at org.kitesdk.data.spi.filesystem.FileSystemMetadataProvider.writeDescriptor(FileSystemMetadataProvider.java:543)
```